karabo_bridge/simulation.py

```diff
@@ -38,39 +38,45 @@ class Detector:
     pixel_size = 0  # [mm]
     distance = 0  # Sample to detector distance [mm]
     layout = np.array([[]])  # super module layout of the detector
-    const_fields = {}
 
     @staticmethod
-    def getDetector(detector, source='', corr=True, gen='random'):
+    def getDetector(detector, source='', raw=False, gen='random'):
         if detector == 'AGIPD':
-            source = source or 'SPB_DET_AGIPD1M-1/DET/detector'
-            return AGIPD(source, corr=corr, gen=gen)
+            if not raw:
+                default = 'SPB_DET_AGIPD1M-1/CAL/APPEND_CORRECTED'
+            else:
+                default = 'SPB_DET_AGIPD1M-1/CAL/APPEND_RAW'
+            source = source or default
+            return AGIPD(source, raw=raw, gen=gen)
         elif detector == 'AGIPDModule':
-            source = source or 'SPB_DET_AGIPD1M-1/DET/0CH0:xtdf'
-            return AGIPDModule(source, corr=corr, gen=gen)
+            source = source or 'SPB_DET_AGIPD1M-1/CAL/0CH0:xtdf'
+            return AGIPDModule(source, raw=raw, gen=gen)
         elif detector == 'LPD':
-            source = source or 'FXE_DET_LPD1M-1/DET/detector'
-            return LPD(source, corr=corr, gen=gen)
+            if not raw:
+                default = 'FXE_DET_LPD1M-1/CAL/APPEND_CORRECTED'
+            else:
+                default = 'FXE_DET_LPD1M-1/CAL/APPEND_RAW'
+            source = source or default
+            return LPD(source, raw=raw, gen=gen)
         else:
             raise NotImplementedError('detector %r not available' % detector)
 
-    def __init__(self, source='', corr=False, gen='random'):
+    def __init__(self, source='', raw=True, gen='random'):
         self.source = source or 'INST_DET_GENERIC/DET/detector'
-        self.corrected = corr
+        self.raw = raw
         if gen == 'random':
             self.genfunc = self.random
         elif gen == 'zeros':
             self.genfunc = self.zeros
         else:
             raise NotImplementedError('gen func %r not implemented' % gen)
 
-    @property
-    def data_shape(self):
-        return (self.modules, self.mod_y, self.mod_x, self.pulses)
-
     @property
     def data_type(self):
-        return np.float32 if self.corrected else np.uint16
+        if self.raw or self.modules == 1:
+            return np.uint16
```

Comment:
This still looks wrong - it should be possible to simulate a single module of corrected data (float32)

Reply:
Yes I missed that